File: 예진희/2 vsStudio_P/210208_cSharp_basic_P/2_class/4_mun1_Samdab/Car.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _4_mun1_Samdab
{
    class Car
    {   // 클래스 상수변수
        public const string DAERI = "동대구 영업소";
        // public: 완전공용 , const : static개념+
        // const string DAERI = "동대구 영업소";

        // 인스턴스 변수 (변수추려내는 거 - 추상화)
        private string company;
        private string color;
        private string price;
        private string model;

        // 기본생성자
        public Car()
        {
            company = "기아";
            color = "블랙";
            model = "K7";
            price = "4천만원";
        }

        // 매개변수가 있는 생성자
        // 생성자 오버로딩
         public Car(string company, string color, string model, string price)
        {
            this.company = company;
            this.color = color;
            this.model = model;
            this.price = price;
        }


        public string Company
        {
            get { return company; }
            set { company = value; }
        }
        public string Model
        {
            get { return model; }
            set { model = value; }
        }
        public string Price
        {
            get { return price; }
            set { price = value; }
        }
        public string Color
        {
            get { return color; }
            set { color = value; }
        }


        public void showCarInfo()
        {
            Console.WriteLine("제조사\t:"+company);
            Console.WriteLine("색상\t:"+color);
            Console.WriteLine("모델\t:"+model);
            Console.WriteLine("가격\t:"+price);
            Console.WriteLine("영업소\t:" + DAERI);
            Console.WriteLine("------------------------\n");
        }
    }
}

File: 황창식/java/class_practice/car_info/src/com/lg/carapp/car/Car.java
package com.lg.carapp.car;

public class Car {
//	static String deri = "동대구 영업소";
//	private String madeCom;
//	private String color;
//	private String model;
//	private String price;
//
//
//	public Car() {
//		madeCom = "기아";
//		model = "K7";
//		price = "4천만원";
//		color = "블랙";
//	}
//
//	public String getMadeCom() {
//		return madeCom;
//	}
//
//	public void setMadeCom(String madeCom) {
//		this.madeCom = madeCom;
//	}
//
//	public String getColor() {
//		return color;
//	}
//
//	public void setColor(String color) {
//		this.color = color;
//	}
//
//	public String getModel() {
//		return model;
//	}
//
//	public void setModel(String model) {
//		this.model = model;
//	}
//
//	public String getPrice() {
//		return price;
//	}
//
//	public void setPrice(String price) {
//		this.price = price;
//	}
//	
//	
//
//	public Car(String madeCom, String color, String model, String price) {
//		super();
//		this.madeCom = madeCom;
//		this.color = color;
//		this.model = model;
//		this.price = price;
//	}
//
//	public void showCarInfo() {
//		System.out.println("제조사 : " + madeCom);
//		System.out.println("색상 : " + color);
//		System.out.println("모델 : " + model);
//		System.out.println("가격 : " + price);
//		System.out.println("대리점명 : " + deri);
//		System.out.println("-------------------");
//	}
	
	public static final String JIJUM = "동대구 영업소";
	private String model;
	private String company;
	private String color;
	private String price;
	
	public Car() {
		model = "K7";
		company = "기아";
		color = "블랙";
		price = "사천만원";
	}

	public Car(String model, String company, String color, String price) {
		super();
		this.model = model;
		this.company = company;
		this.color = color;
		this.price = price;
	}
	
	public String getModel() {
		return model;
	}

	public void setModel(String model) {
		this.model = model;
	}

	public String getPrice() {
		return price;
	}

	public void setPrice(String price) {
		this.price = price;
	}

	public void showCarInfo() {
		System.out.println("제조사	: " + company);
		System.out.println("모델명	: " + model);
		System.out.println("색상	: " + color);
		System.out.println("가격	: " + price);
		System.out.println("대리점명 : " + JIJUM);
		System.out.println("---------------------------");
	}
}
